### ISSUE TITLE
jest unit test is broken after upgrade @preact/signals from 1.3.1 to 2.0.0.

### ISSUE THREAD

**Issue by rock-57blocks** [preactjs/signals#633]
- [x] Check if updating to the latest version resolves the issue

**Environment**
- [ ] I am using `@preact/signals-core`
- [x] I am using `@preact/signals`
- [ ] I am using `@preact/signals-react`
  - [ ] I understand usage changed in v2, and I've followed the [React Integration instructions](https://github.com/preactjs/signals/tree/main/packages/react#react-integration)

**Describe the bug**
jest unit test is broken after upgrading `@preact/signals` from 1.3.1 to 2.0.0.
Follow the instructions for release updates: `When you perform a signal update, you'll need to wrap it in act.`
I did it, but it still failed. This test can pass after rolling back to 1.3.1

**To Reproduce**
https://stackblitz.com/~/github.com/rock-57blocks/NeverReact

1. Go to [stackblitz](https://stackblitz.com/~/github.com/rock-57blocks/NeverReact)
2. open terminal
3. run 'npm test'

**Expected behavior**
This test is passed on @preact/signals 1.3.1, but failed on 2.0.0.

**Comment by rock-57blocks** [comment]
I found that if a child component uses `useEffect`, and its dependencies have prop `signal value` ​​from the parent component, it will make the parent component's `useSignalEffect callback invalid`.

e.g

**Comment by JoviDeCroock** [comment]
Hmm yes, looks like the signal-effect does not rerun when the dependency changes

**Comment by rock-57blocks** [comment]
> Hmm yes, looks like the signal-effect does not rerun when the dependency changes - does it work at runtime?
> 

@JoviDeCroock Thanks for the quick replay. 
I tested it and it works fine at runtime. This issue only occurs during unit testing.

BTW,  what confuses me is that if the child component's `useEffect` does not depend on the signal parameter in the signal props but only on something else, then this test will work.

See the follow code:

```
const MuteButton: FunctionalComponent<MuteButtonProps> = ({
  isMuted,
  isTestMuted,
}) => {
  useEffect(() => {
    console.log('isMuted', isTestMuted);
  }, [isTestMuted]);

  return null;
};

export const muteVideo = signal(false);
const TestPlayer: FunctionalComponent = () => {
  const videoRef = useRef<HTMLVideoElement>(null);
  useSignalEffect(() => {
    videoRef.current!.muted = muteVideo.value;
  });

  return (
    <Fragment>
      <div>
        <video
          data-testid='video-player'
          ref={videoRef}>
        </video>
        <MuteButton
          isMuted={muteVideo.value}
          isTestMuted={true}
          />
      </div>
    </Fragment>
  );

};

```

![Screenshot 2024-12-25 at 18 06 34](https://github.com/user-attachments/assets/bd9ac6a0-405f-4395-b48d-2d3488806686)

**Comment by JoviDeCroock** [comment]
Yes, the act override all explains that. We queue twice because the timing of rerender --> effect queued up, signal update --> effect queued up are slightly different, see my last edits to the above which includes the fix

**Comment by rock-57blocks** [comment]
I see..
Will it be include in a new version 2.0.1 very soon or later？ 
Thanks a lot. @JoviDeCroock

**Comment by JoviDeCroock** [comment]
As I said, I have no time now... it's Christmas today 😅 I gave you the solution so you can PR or patch it if it's urgent, this is from my phone

**Comment by rock-57blocks** [comment]
oh sorry. I'm not rushing you just to ask about a possible release date, don't worry it's not a very urgent need, please enjoy your holidays and Merry Christmas！

**Comment by JoviDeCroock** [comment]
This will be fixed in 10.25.4, thanks to @rock-57blocks! You can track https://github.com/preactjs/preact/pull/4620 to see when the release goes out

**Comment by JoviDeCroock** [comment]
Released in 10.25.4
